File: CaseStudies/FileInputTests/Bundled/Design4/main.jl
# Bundled

#=
    This is actually super elegant, at least for this size.
    the issue is that it could make the constructor really long if we do it this
    way.  The good news is, Designs 5 and 6 have similar constraints that are
    still modular.
=#

# Struct for holding parameters.
struct Parameters
    name::AbstractString
    height::Integer
    weight::AbstractFloat
end

function Parameters(filename::AbstractString)
    
    # get_input
    file = open(filename, "r")
    lines = readlines(file)
    close(file)

    name = lines[2]
    height = parse(Int32, lines[4])
    weight = parse(Float64, lines[6])

    # input_constraints
    if isempty(strip(name))
        throw("Name must not be blank!")
    elseif height <= 0
        throw("Height must be positive!")
    elseif weight <= 0
        throw("Weight must be positive!")
    end

    return Parameters(name, height, weight)
end

p = Parameters("FileInputTests/input.txt")
println(p)

File: CaseStudies/FileInputTests/Bundled/Design4/main.swift
// Bundled

import Foundation

// Class for holding parameters
class Parameters {
    var name: String // Default values not needed now!
    var height: Int
    var weight: Double

    init(_ filename: String) throws {

        // get_input
        let lines = try String(contentsOfFile: filename, encoding: String.Encoding.utf8)
            .components(separatedBy: "\n")
        
        self.name = lines[1]
        self.height = Int(lines[3]) ?? 0
        self.weight = Double(lines[5]) ?? 0
        
        // input_constraints
        if self.name.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
            throw ParamError("Name must not be blank!")
        }
        else if self.height <= 0 {
            throw ParamError("Height must be positive!")
        }
        else if self.weight <= 0 {
            throw ParamError("Weight must be positive!")
        }
    }
}

// Simple error struct
struct ParamError: LocalizedError {
    let description: String

    init(_ description: String) {
        self.description = description
    }

    var errorDescription: String? {
        description
    }
}

var p: Parameters = try Parameters("FileInputTests/input.txt")
print("Parameters(\"" + p.name + "\", " + String(p.height) + ", " + String(p.weight) + ")")